File: OneCodeTeam/How-to-log-UI-traces-into-Windows-Azure-table-29443/[VB]-How to log UI traces into Windows Azure table storage/C#/CSAzureJSLogging/Controllers/HomeController.cs
/****************************** Module Header ******************************\
* Module Name: HomeController.cs
* Project:     CSAzureJSLogging
* Copyright (c) Microsoft Corporation
*
* HomeController
* 
* This source is subject to the Microsoft Public License.
* See http://www.microsoft.com/opensource/licenses.mspx#Ms-PL.
* All other rights reserved.
* 
* THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
* EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
* WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
\*****************************************************************************/
using System.Web.Mvc;

namespace CSAzureJSLogging.Controllers
{
     
    public class HomeController : Controller
    {   
        // GET: Home
        public ActionResult Index()
        {
            return View();
        }

    }
}

File: OneCodeTeam/How-to-construct-data-to-json-string-before-27410/[VB]-How to construct data to json string before posting to pagemethod using jQuery/VB/VBASPNETUseJsonString/Controllers/HomeController.vb
'***************************** Module Header ******************************\
' Module Name:	HomeController.vb
' Project:		VBASPNETUseJsonString
' Copyright (c) Microsoft Corporation.
' 
' This sample shows how to post complex data to MVC server side using json string.
' 
' This source is subject to the Microsoft Public License.
' See http://www.microsoft.com/en-us/openness/licenses.aspx#MPL.
' All other rights reserved.
' 
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED 
'WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
'**************************************************************************/
Imports Newtonsoft.Json

Public Class HomeController
    Inherits System.Web.Mvc.Controller

    '
    ' GET: /Home
    Public Function Index() As ActionResult
        Return View()
    End Function

    <HttpPost> _
    Public Function saveUserInfos(data As String) As ActionResult
        Try
            Dim userInfoList = JsonConvert.DeserializeObject(Of IEnumerable(Of UserInfo))(data)
            'Save these data to Your DB
            Return Json("Data saved successfuly.")
        Catch e As JsonReaderException
            Return Json(e.Message)
        End Try
    End Function


End Class